swift_scality_backend/server.py

```diff
@@ -48,10 +48,20 @@ def setup(self, conf):
         sproxyd_urls = ['http://%s/%s/' % (h, sproxyd_path) for h in
                         swift_scality_backend.utils.split_list(conf['sproxyd_host'])]
 
-        self._filesystem = SproxydClient(sproxyd_urls,
-                                         conf.get('sproxyd_conn_timeout'),
-                                         conf.get('sproxyd_proxy_timeout'),
-                                         self.logger)
+        # We can't pass `None` as value for sproxyd_*_timeout because it will
+        # override the defaults set in SproxydClient
+        kwarg = {}
+        try:
+            kwarg['sproxyd_conn_timeout'] = float(conf.get('sproxyd_conn_timeout'))
+        except TypeError:
```

Comment:
I guess a `TypeError` is only raised when `conf.get` returns `None`, but I'd be a bit more specific then, something along the lines of

```python
sproxyd_conn_timeout = conf.get('sproxyd_conn_timeout')
if sproxyd_conn_timeout is not None:
    kwargs['sproxyd_conn_timeout'] = float(sproxyd_conn_timeout)
```
and similar for `sproxyd_proxy_timeout`, or even go further:

```python
def get_config(name, fun):
    value = conf.get(name)
    return fun(value) if value is not None else None

kwargs = {
    'sproxyd_conn_timeout': get_config('sproxyd_conn_timeout', float),
    'sproxyd_proxy_timeout': get_config('sproxyd_proxy_timeout', float),
}
```